File: En Language Python/5-Rapide.py

def partitionner(tableau, bas, haut):
    pivot = tableau[haut]
    i = bas - 1
    for j in range(bas, haut):
        if tableau[j] <= pivot:
            i += 1
            tableau[i], tableau[j] = tableau[j], tableau[i]
    tableau[i + 1], tableau[haut] = tableau[haut], tableau[i + 1]
    return i + 1

def tri_rapide(tableau, bas, haut):
    if bas < haut:
        pivot = partitionner(tableau, bas, haut)
        tri_rapide(tableau, bas, pivot - 1)
        tri_rapide(tableau, pivot + 1, haut)

# Exemple d'utilisation
tableau = [64, 25, 12, 22, 11]
tri_rapide(tableau, 0, len(tableau) - 1)
print("Tableau trié par rapide:", tableau)

File: En Language C/5-Rapide.c

//Tri Rapide (Quick Sort)
//Complexité : O(n²)

void triRapide(int T[], int bas, int haut)
{
    if (bas < haut)
    {
        int pi = partition(T, bas, haut);
        triRapide(T, bas, pi - 1);
        triRapide(T, pi + 1, haut);
    }
}
int partition(int T[], int bas, int haut)
{
    int pivot = T[haut];
    int i = (bas - 1);
    for (int j = bas; j < haut; j++)
    {
        if (T[j] <= pivot)
        {
            i++;
            echanger(&T[i], &T[j]);
        }
    }
    echanger(&T[i + 1], &T[haut]);
    return (i + 1);
}

void echanger(int *a, int *b)
{
    int tmp = *a;
    *a = *b;
    *b = tmp;
}